hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -282,6 +284,9 @@
 
   private HeapMemoryManager hMemManager;
 
+  private volatile boolean compactionOffloadEnabled = false;
+  private SplitOrMergeTracker switchTracker;
```

Comment:
We do not use this field at region server side in the past? Then I think we need to consider whether this is the correct approach.

I checked the design doc, not too much information about this. Could you please explain the design more?

Thanks.